source/common/router/router.cc

```diff
@@ -1399,6 +1411,13 @@ void Filter::UpstreamRequest::onPoolReady(Http::StreamEncoder& request_encoder,
                                           Upstream::HostDescriptionConstSharedPtr host) {
   ENVOY_STREAM_LOG(debug, "pool ready", *parent_.callbacks_);
 
+  // Send value zero for optional HTTP code. It is invalid code and means that
+  // LOCAL_ORIGIN_CONNECT_SUCCESS is not mapped to HTTP code when running in outlier non-split mode.
+  // If it was mapped to HTTP code it would interfere with code returned by HTTP server and two
+  // codes would be reported to outlier detector for one request.
+  host->outlierDetector().putResult(Upstream::Outlier::Result::LOCAL_ORIGIN_CONNECT_SUCCESS,
+                                    absl::optional<uint64_t>(0));
```

Comment:
Would it be better to have a different putResult API that better captures this case? Sending 0 here seems like a hack to me. Why can't we just send nullopt, because that is the default? If so, maybe another API makes sense? It would be nice to match this up somehow w/ the LOCAL_ORIGIN_CONNECT_FAILED call above.